File: tools/sub.py
import time
import struct
import zmq

def deserialize(data):
    return {
        'topic': data[0],
        'serial': data[1],
        'timestamp': struct.unpack('d', data[2])[0],
        'data': data[3],
        'setbits': data[4],
        'zscore': struct.unpack('d', data[5])[0]
    }

context = zmq.Context.instance()
sub = context.socket(zmq.SUB)
sub.connect('ipc:///tmp/backend')
#sub.connect('tcp://127.0.0.1:5000')
sub.setsockopt(zmq.SUBSCRIBE, 'entropy'.encode('utf-8'))

while True:
    try:
        while True:
            packet = sub.recv_serialized(deserialize, zmq.NOBLOCK)
            print(packet)
    except zmq.ZMQError:
        pass # No more data
    time.sleep(1)

File: tools/sub.c
// gcc -lzmq -lczmq -o sub sub.c

#include <czmq.h>

int main(void) {

    zsock_t *sub = zsock_new_sub("ipc:///tmp/backend", "entropy");
    //zsock_t *sub = zsock_new_sub("tcp://127.0.0.1:5000", "entropy");
    size_t size_double, size_data;
    char *topic, *serial;
    byte *timestamp_bytes, *z_bytes, *data;
    double timestamp, z;
    int setbits;

    while (!zsys_interrupted) {
        zsock_recv(sub, "ssbbib", &topic, &serial, &timestamp_bytes, &size_double, &data, &size_data, &setbits, &z_bytes, &size_double);
        memcpy(&timestamp, timestamp_bytes, sizeof(double));
        memcpy(&z, z_bytes, sizeof(double));
        printf("%s - %s - %lf - %s - %d - %lf\n", topic, serial, timestamp, data, setbits, z);
        free(topic);
        free(serial);
        free(timestamp_bytes);
        free(z_bytes);
        free(data);
    }

    zsock_destroy(&sub);

    return 0;
}